### ISSUE TITLE
SysPFMG could not work

### ISSUE THREAD

**Issue by zongy17** [hypre-space/hypre#822]
When I try solving my linear system by SysPFMG through SStruct interface, I got such runtime errors reported by hypre:

`[cn002:287395:0:287395] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x378)
==== backtrace ====
 0 0x000000000045431e hypre_PFMGComputeDxyz_SS27()  /home/users/zongyi/software-package/hypre-2.25.0/src/struct_ls/pfmg_setup.c:2108
 1 0x000000000044e980 hypre_PFMGComputeDxyz()  /home/users/zongyi/software-package/hypre-2.25.0/src/struct_ls/pfmg_setup.c:824
 2 0x000000000040621f hypre_SysPFMGSetup()  /home/users/zongyi/software-package/hypre-2.25.0/src/sstruct_ls/sys_pfmg_setup.c:159
 3 0x0000000000404245 HYPRE_SStructSysPFMGSetup()  /home/users/zongyi/software-package/hypre-2.25.0/src/sstruct_ls/HYPRE_sstruct_sys_pfmg.c:39
 4 0x00000000004a7384 hypre_PCGSetup()  /home/users/zongyi/software-package/hypre-2.25.0/src/krylov/pcg.c:236
 5 0x00000000004a678d HYPRE_PCGSetup()  /home/users/zongyi/software-package/hypre-2.25.0/src/krylov/HYPRE_pcg.c:34
 6 0x0000000000403de9 HYPRE_SStructPCGSetup()  /home/users/zongyi/software-package/hypre-2.25.0/src/sstruct_ls/HYPRE_sstruct_pcg.c:51
 7 0x00000000004037a0 main()  /home/users/zongyi/HUAWEI/baseline/base_sstruct.c:379
 8 0x00000000000223d5 __libc_start_main()  ???:0
 9 0x0000000000403bb7 _start()  ???:0
===================
srun: error: cn002: task 0: Segmentation fault (core dumped)
`

I used SysPFMG to solve a 3d7 linear system successfully before. But this time when I tried 3d27, things went bad. The code at struct_ls/pfmg_setup.c:2108 is computing tcx/tcy/tcz. Does anyone have any iead about it?

**Comment by zongy17** [comment]
After I assembled the SStructVector b and SStructMatrix A, I called functions hypre_SStructKrylovInnerProd and HYPRE_SStructMatrixMatvec to see the results of (b, b) and (A*b, A*b). The results showed good and is consistent with the result given by Matlab and other programs, so I think my code to call hypre is correct.
But it ran into segfault at PFMG setup process.

**Comment by zongy17** [comment]
Finally I fixed it. It was a bug in my code to define stencil offsets. There were 27 group of offsets in total, each consists of 3 int. I provided two same offsets within 27 as a mistake, and thus neglected another one in 27. HYPRE treated it as 3d27 pattern, and accessed its pointers through funtion hypre_StructMatrixExtractPointerByIndex. The function call corresponding to the neglected one would return a NULL pointer, and cause following segfault when HYPRE tried accessing values inside this pointer.
